solidity/dashboard/src/services/staking-port-backer.service.js

```diff
@@ -0,0 +1,193 @@
+    filterOutByOperator(operatorsToSkip)
+  )
+  const filteredManagedGrantOperators = managedGrantOperators.filter(
+    filterOutByOperator(operatorsToSkip)
+  )
+
+  const ownedDelegations = await getDelegations(
+    filteredOwnedOperators,
+    undelegationPeriod
+  )
+
+  const granteeDelegations = await getDelegations(
+    filteredGranteeOperators,
+    undelegationPeriod,
+    true
+  )
+
+  const managedGrantsDelegations = await getDelegations(
+    filteredManagedGrantOperators,
+    undelegationPeriod,
+    true,
+    true
+  )
+
+  const delegations = [
+    ...ownedDelegations,
+    ...granteeDelegations,
+    ...managedGrantsDelegations,
+  ].sort((a, b) => sub(b.createdAt, a.createdAt))
+
+  return {
+    delegations,
+    undelegationPeriod,
+  }
+}
+
+const getDelegations = async (
+  operators,
+  undelegationPeriod,
+  isFromGrant,
+  isManagedGrant
+) => {
+  const web3 = await Web3Loaded
+  const { oldTokenStakingContract, grantContract } = await ContractsLoaded
+
+  const delegations = []
+
+  for (const operatorAddress of operators) {
+    const {
+      createdAt,
+      undelegatedAt,
+      amount,
+    } = await oldTokenStakingContract.methods
+      .getDelegationInfo(operatorAddress)
+      .call()
+
+    const beneficiary = await oldTokenStakingContract.methods
+      .beneficiaryOf(operatorAddress)
+      .call()
+    const authorizerAddress = await oldTokenStakingContract.methods
+      .authorizerOf(operatorAddress)
+      .call()
+
+    let grantId = null
+    let managedGrantContractInstance = null
+    if (isFromGrant) {
+      try {
+        const grantStakeDetails = await grantContract.methods
+          .getGrantStakeDetails(operatorAddress)
+          .call()
+        grantId = grantStakeDetails.grantId
+      } catch (error) {
+        grantId = null
+      }
+    }
+
+    if (isManagedGrant && grantId) {
+      const { grantee } = await grantContract.methods.getGrant(grantId).call()
+      managedGrantContractInstance = createManagedGrantContractInstance(
+        web3,
+        grantee
+      )
+    }
+
+    const operatorData = {
+      undelegatedAt,
+      amount,
+      beneficiary,
+      operatorAddress,
+      createdAt,
+      authorizerAddress,
+      isFromGrant,
+      grantId,
+      isManagedGrant,
+      managedGrantContractInstance,
+    }
+    const balance = web3Utils.toBN(amount)
+
+    if (!balance.isZero() && operatorData.undelegatedAt === "0") {
+      operatorData.isUndelegation = false
+      delegations.push(operatorData)
+    }
+
+    if (operatorData.undelegatedAt !== "0" && gt(amount, 0)) {
+      operatorData.undelegationCompleteAt = moment
+        .unix(undelegatedAt)
+        .add(undelegationPeriod, "seconds")
+      operatorData.canRecoverStake = operatorData.undelegationCompleteAt.isBefore(
+        moment()
+      )
+      operatorData.isUndelegation = true
+      delegations.push(operatorData)
+    }
```

Comment:
`balance` is the delegation `amount` converted to BN, right?
In the first, `if` statement we check if the amount is different than zero and in the second we check if it's greater than zero. Thoughts on updating it as in the suggestion below?

Do we even have to check the delegation amount to be greater than zero, probably yes, because `StakingPortBacker` requires that.

```suggestion
    const delegationAmount = web3Utils.toBN(amount)

    // StakingPortBacker contract requires the delegation amount to be greater than zero.
    if (delegationAmount.gt(0)) {
      // Check if the stake is currently undelegating.
      if (operatorData.undelegatedAt === "0") {
        operatorData.isUndelegation = false
      } else {
        operatorData.undelegationCompleteAt = moment
          .unix(undelegatedAt)
          .add(undelegationPeriod, "seconds")
        operatorData.canRecoverStake = operatorData.undelegationCompleteAt.isBefore(
          moment()
        )
        operatorData.isUndelegation = true
      }
      
      delegations.push(operatorData)
    }
```

Reply:
pls see 7a9c6c3